hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheEventJournalSubscribeTask.java

```diff
@@ -0,0 +1,93 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.cache;
+
+import com.hazelcast.cache.impl.CacheService;
+import com.hazelcast.cache.impl.journal.CacheEventJournalSubscribeOperation;
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.CacheEventJournalSubscribeCodec;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.cluster.Versions;
+import com.hazelcast.journal.EventJournalInitialSubscriberState;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.CachePermission;
+import com.hazelcast.spi.Operation;
+
+import java.security.Permission;
+
+
+/**
+ * Performs the initial subscription to the cache event journal.
+ * This includes retrieving the event journal sequences of the
+ * oldest and newest event in the journal.
+ *
+ * @see com.hazelcast.cache.impl.CacheProxy#subscribeToEventJournal
+ * @see CacheEventJournalSubscribeOperation
+ * @since 3.9
+ */
+public class CacheEventJournalSubscribeTask
+        extends AbstractCacheMessageTask<CacheEventJournalSubscribeCodec.RequestParameters> {
+
+    public CacheEventJournalSubscribeTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Operation prepareOperation() {
+        if (nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V3_9)) {
+            throw new UnsupportedOperationException(
+                    "Event journal actions are not available when cluster version is 3.9 or higher");
+        }
+        return new CacheEventJournalSubscribeOperation(parameters.name);
+    }
+
+    @Override
+    protected CacheEventJournalSubscribeCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return CacheEventJournalSubscribeCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        final EventJournalInitialSubscriberState state = (EventJournalInitialSubscriberState) response;
+        return CacheEventJournalSubscribeCodec.encodeResponse(state.getOldestSequence(), state.getNewestSequence());
+    }
+
+    @Override
+    public final String getServiceName() {
+        return CacheService.SERVICE_NAME;
+    }
+
+    public Permission getRequiredPermission() {
+        return new CachePermission(parameters.name, ActionConstants.ACTION_READ);
```

Comment:
Different type of action, maybe subscribe ? I believe we have something similar in the listeners